### PR TITLE
chore(eslint-plugin-internal): [plugin-test-formatting] support random object literal tests

### PR DESCRIPTION
**Pull request by bradzacher**
## Overview

I was looking at some of our tests and noticed they weren't formatted correctly.
Investigating it was because they were declared as variables instead of as part of a test run.

This PR just adds support to the rule for random object literals by doing some fuzzy checks using types.

### CODE REVIEW

File: packages/eslint-plugin-internal/src/rules/plugin-test-formatting.ts [hunk 1]
```diff
@@ -478,33 +491,99 @@ export default createRule<Options, MessageIds>({
...
       // invalid - errors
-      [invalidTestsSelectorPath.join(' > ')]: checkInvalidTest,
-      // invalid - suggestions
       [[
-        ...invalidTestsSelectorPath,
-        'Property[key.name = "errors"]',
-        AST_NODE_TYPES.ArrayExpression,
+        AST_NODE_TYPES.CallExpression,
         AST_NODE_TYPES.ObjectExpression,
-        'Property[key.name = "suggestions"]',
```
Comment (bradzacher) [inline packages/eslint-plugin-internal/src/rules/plugin-test-formatting.ts:501]: in #5379 we stopped checking output formatting.
suggestions only ever have output code, so no sense even bothering to check them!

File: packages/eslint-plugin-internal/src/rules/plugin-test-formatting.ts [hunk 2]
```diff
@@ -478,33 +491,99 @@ export default createRule<Options, MessageIds>({
...
+
+        if (/^InvalidTestCase\b/.test(typeString)) {
+          checkInvalidTest(node);
+          for (const testProp of node.properties) {
+            if (
+              testProp.type === AST_NODE_TYPES.SpreadElement ||
+              testProp.computed ||
+              testProp.key.type !== AST_NODE_TYPES.Identifier ||
+              testProp.key.name !== 'errors' ||
+              testProp.value.type !== AST_NODE_TYPES.ArrayExpression
```
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin-internal/src/rules/plugin-test-formatting.ts:572]: Nit: this is _almost_ the same as the previous `.type === AST_NODE_TYPES.SpreadElement || ...` check, but for the `'errors'` case. I wonder if there's a good way to dedup? 🤷 definitely not a blocker IMO.

**Review by JoshuaKGoldberg — APPROVED** [review]
Super, thanks! ✨

File: packages/eslint-plugin/tests/rules/member-ordering/member-ordering-alphabetically-case-insensitive-order.test.ts [hunk 3]
```diff
@@ -1,23 +1,22 @@
...
 import rule, { defaultOrder } from '../../../src/rules/member-ordering';
+import type { RunTests } from '../../RuleTester';
 import { RuleTester } from '../../RuleTester';
 
 const ruleTester = new RuleTester({
   parser: '@typescript-eslint/parser',
 });
 
-const sortedCiWithoutGrouping: TSESLint.RunTests<MessageIds, Options> = {
+const sortedCiWithoutGrouping: RunTests<MessageIds, Options> = {
```
[the remark this answers is not in the document]
Comment (bradzacher) [reply packages/eslint-plugin/tests/rules/member-ordering/member-ordering-alphabetically-case-insensitive-order.test.ts:10]: I'm personally not too miffed either way.

I think the separate variables is nice as a way to document and group the tests.
it's much clearer than the alternative:
```ts
ruleTester.run('member-ordering', rule, {
  valid: [
    //
    // Sorted case insensitive without grouping
    //
    'test1',
    // ...

    //
    // Sorted case insensitive with grouping
    //
    'testn+1',
    // ...
  ],
});
```
I find it really easy to lose the comments in the array when it gets really big.

But the indirection is a bit of a pain overall.

I think either way has trade-offs, so IDK which one we want as the standard in the repo.

[the remark this answers is not in the document]
Comment (JoshuaKGoldberg) [reply packages/eslint-plugin/tests/rules/member-ordering/member-ordering-alphabetically-case-insensitive-order.test.ts:10]: My mild preference would be for less code, but it's not a hill I'm going to die on 😄

**Pull request merged**